module/zfs/zil.c

```diff
@@ -1179,7 +1179,8 @@ zil_lwb_flush_vdevs_done(zio_t *zio)
 		ASSERT3P(zcw->zcw_lwb, ==, lwb);
 		zcw->zcw_lwb = NULL;
 
-		zcw->zcw_zio_error = zio->io_error;
+		if (zio->io_error != 0)
+			zcw->zcw_zio_error = zio->io_error;
```

Comment:
There should be a comment here explaining why we need to do this check.
Also, we should VERIFY that `zcw->zcw_zio_error == 0` before overwriting it with `zio->io_error`.
IIUC, we can assert that because, IIUC, we don't issue the flush if the write fails.
We should actually `VERIFY` because
a) it's not on a hot code path and
b) it's critical for correctness.

The comment should address the assertion as well.